### PR TITLE
Two new preprocessing modules

### PR DESCRIPTION
**Pull request by mhhennig**
- BlankSaturationRecording finds extreme outliers in data and blanks these. Useful for Biocam recordings.
- ClipTracesRecording allows clipping the data
- NormalizeByQuantileRecording and BlankSaturationRecording now use a fixed random seed for identical runs.

Would be great to have these on PyPi soon, as I want to distribute some code using these.

### CODE REVIEW

**Comment by alejoe91** [comment]
@mhhennig Can you merge with master and commit? I think this is not up to date and that's why the tests fail

File: spiketoolkit/preprocessing/blank_saturation.py [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+            traces[traces>=self._threshold] = self._median
+        return traces
+
+
+def blank_saturation(recording, threshold=None):
+    '''
+    Find and remove parts of the signla with extereme values. Some arrays
+    may produce these when amplifiers enter saturation, typically for
+    short periods of time. To remove these artefacts, values below or above 
+    a threshold are set to the medina signal value.
```
Comment (alejoe91) [inline spiketoolkit/preprocessing/blank_saturation.py]: medina -> median

**Comment by colehurwitz** [comment]
Whoops, added curation docs to this. Not a big deal though just some rst stuff.

**Comment by alejoe91** [comment]
@colehurwitz ok! but in general it's better to keep PR separate :) as master @samuelgarcia says

**Comment by colehurwitz** [comment]
Agreed! Much better practice to keep it separated. Better merge fast so no one notices ;)

**Comment by colehurwitz** [comment]
Can I merge?

**Pull request merged**